launcher.py

```diff
@@ -12,94 +12,149 @@
 import webbrowser
 from importlib.metadata import PackageNotFoundError, version
 from pathlib import Path
+from typing import Optional
 
-CONFIG_FILE = Path("config.json")
-RUNTIME_CONFIG_FILE = Path("runtime_config.json")
-VENV_NAME = "virtual_env"  # For downloading python packages
-VENV_DIR = Path.cwd() / VENV_NAME  # Generate the path for the virtual environment
 MEDIAPROCESSOR_PATH = Path("MediaProcessor") / "build"
-DEFAULT_CONFIG = {
-    "version": 1,
-    "formatters": {
-        "detailed": {"format": "%(asctime)s [%(levelname)s] [%(funcName)s(), %(lineno)d]: %(message)s"},
-        "simple": {"format": "[%(levelname)s] %(message)s"},
-    },
-    "handlers": {
-        "console": {
-            "class": "logging.StreamHandler",
-            "formatter": "simple",
-            "stream": "ext://sys.stdout",
+LOG_LOCK = threading.Lock()
+
+
+class Config:
```

Comment:
From what I see this isn't in itself a Config, but rather something that modifies and manages configurations, be it files or processes. In these cases the name should reflect this. Please rename it to `ConfigManager`.